### PR TITLE
[SW-1089] Account screen for BTC

### PR DESCRIPTION
**Pull request by konradodo**

### CODE REVIEW

File: src/popup/components/TransactionList.vue [hunk 1]
```diff
@@ -221,11 +226,16 @@ export default defineComponent({
...
+        if (activeAccount.value.protocol === PROTOCOL_AETERNITY) {
+          await fetchTransactions({
+            limit: TXS_PER_PAGE,
+            recent,
+            address: currentAddress.value,
+          });
+        }
+        if (activeAccount.value.protocol === PROTOCOL_BITCOIN) {
+          // TODO fetch Bitcoin transactions
+        }
```
Comment (pstachel) [inline src/popup/components/TransactionList.vue:238]: I'd suggest to handle this by protocol adapter

File: src/protocols/bitcoin/views/AccountDetails.vue [hunk 2]
```diff
@@ -1,19 +1,27 @@
...
 
     <template #buttons>
-      <BtnBox
-        text="Example button 1"
-      />
-      <BtnBox
-        text="Example button 2"
-      />
+      <OpenTransferReceiveModalButton />
+      <OpenTransferSendModalButton />
```
Comment (pstachel) [inline src/protocols/bitcoin/views/AccountDetails.vue:24]: I think those could be handled by the base component as the logic is handled elsewhere

File: src/protocols/bitcoin/views/AccountDetails.vue [hunk 3]
```diff
@@ -1,19 +1,27 @@
...
     <template #balance>
-      <div class="text-center">
-        <h3>BTC</h3>
-        <p>This is a temporary component that uses the account base boilerplate.</p>
-      </div>
+      <BalanceInfo
+        :balance="balanceNumeric"
+        :protocol="activeAccount.protocol"
+        horizontal-offline-message
+      />
```
Comment (pstachel) [inline src/protocols/bitcoin/views/AccountDetails.vue:19]: Couldn't that stay in base component - I guess there's no protocol specific properties here

File: src/protocols/bitcoin/views/AccountDetails.vue [hunk 4]
```diff
@@ -23,20 +31,51 @@
...
 import AccountDetailsBase from '@/popup/components/AccountDetailsBase.vue';
-import BtnBox from '@/popup/components/buttons/BtnBox.vue';
+import AccountInfo from '@/popup/components/AccountInfo.vue';
+import { useAccounts, useBalances } from '@/composables';
+import { useStore } from 'vuex';
+import BalanceInfo from '@/popup/components/BalanceInfo.vue';
+import { useRoute, useRouter } from 'vue-router';
+import { ROUTE_ACCOUNT_DETAILS_TRANSACTIONS } from '@/popup/router/routeNames';
+import OpenTransferReceiveModalButton from '@/popup/components/OpenTransferReceiveModalButton.vue';
+import OpenTransferSendModalButton from '@/popup/components/OpenTransferSendModalButton.vue';
```
Comment (pstachel) [inline src/protocols/bitcoin/views/AccountDetails.vue:45]: Please organise the imports

File: src/protocols/bitcoin/views/AccountDetails.vue [hunk 5]
```diff
@@ -23,20 +31,51 @@
...
+    const { balance } = useBalances({ store });
+
+    const balanceNumeric = computed(() => balance.value.toNumber());
+
+    (() => {
+      // BTC have only one view in ACCOUNT DETAILS
+      if (route.name !== ROUTE_ACCOUNT_DETAILS_TRANSACTIONS) {
+        router.push({ name: ROUTE_ACCOUNT_DETAILS_TRANSACTIONS });
+      }
+    })();
```
Comment (pstachel) [inline src/protocols/bitcoin/views/AccountDetails.vue:74]: Why is it needed - isn't it handled by the router?

**Review by pstachel — CHANGES_REQUESTED** [review]

File: src/protocols/bitcoin/views/AccountDetails.vue (hunk 5 above)
Comment (konradodo) [reply src/protocols/bitcoin/views/AccountDetails.vue]: AccountDetails is wrapper for other routes and BTC have only one of them.

File: src/popup/components/AccountDetailsBase.vue [hunk 6]
```diff
@@ -52,24 +63,34 @@ import {
...
+    OpenTransferSendModalButton,
+    OpenTransferReceiveModalButton,
     TransactionAndTokenFilter,
     BtnClose,
   },
   setup() {
     const route = useRoute();
-
+    const store = useStore();
+    const { activeAccount } = useAccounts({ store });
```
Comment (peronczyk) [inline src/popup/components/AccountDetailsBase.vue:93]: Please add empty line after this one.

**Review by pstachel — APPROVED** [review]

File: src/popup/components/AccountDetailsBase.vue [hunk 7]
```diff
@@ -4,17 +4,39 @@
     class="account-details"
   >
     <div class="account-info-wrapper">
-      <slot name="account-info" />
-
+      <slot
+        v-if="$slots['account-info']"
+        name="account-info"
+      />
```
Comment (peronczyk) [inline src/popup/components/AccountDetailsBase.vue:10]: It's not used so remove for now.

File: src/protocols/bitcoin/views/AccountDetails.vue [hunk 8]
```diff
@@ -1,42 +1,39 @@
...
   components: {
     AccountDetailsBase,
-    BtnBox,
   },
+
   setup() {
+    const store = useStore();
+    const {
+      activeAccount,
+    } = useAccounts({ store });
```
Comment (peronczyk) [inline src/protocols/bitcoin/views/AccountDetails.vue:24]: No need to split this line.

File: src/protocols/bitcoin/views/AccountDetails.vue [hunk 9]
```diff
@@ -1,42 +1,39 @@
...
+    const route = useRoute();
+    const router = useRouter();
```
Comment (peronczyk) [inline src/protocols/bitcoin/views/AccountDetails.vue:26]: Please put this before local composables (accounts).

File: src/protocols/bitcoin/views/AccountDetails.vue [hunk 10]
```diff
@@ -1,42 +1,39 @@
...
+
+    (() => {
+      // BTC have only one view (transaction list) in ACCOUNT DETAILS
+      if (route.name !== ROUTE_ACCOUNT_DETAILS_TRANSACTIONS) {
+        router.push({ name: ROUTE_ACCOUNT_DETAILS_TRANSACTIONS });
+      }
+    })();
+
     return {
+      activeAccount,
```
Comment (peronczyk) [inline src/protocols/bitcoin/views/AccountDetails.vue:36]: This is not used within the template.

**Review by peronczyk — APPROVED** [review]

**Pull request merged**